### ISSUE TITLE
Uncaught ReferenceError: emotionReact is not defined at 9291 (userCodeAppPanel:2:53729)  

### ISSUE THREAD

**Issue by iamsanteri** [enuchi/React-Google-Apps-Script#215]
Hi, I've run into problems when using the command to bundle and deploy the app with: "npm run deploy". My application works fully well in the development environment with "npm run start", but when deploying and trying to run the editor add-on extension in non-development mode the sidebar remains on a white screen and console complains of emotionReact not being defined. Apps Script executions and logs show no errors whatsoever. 

After trying to debug the issue for some time now and downgrading emotionReact package in package.json, as well as trying to get to the source of the issue I'm frankly just lost at where or when this error got introduced and why. I simply cannot find the point in time that introduced this error. The problem is also that I cannot see where or what is causing this as it seems to come from userCodeAppPanel which opens into a blank view in the console (see pictures below). 

Any idea of how I could best start finding out what is going on or what to look for trying to debug this? It's really unsettling to me when my development (which works fine) and an eventual deploy into production diverges like this and I'm really quite lost here on how to get back to a working production deploy. Any ideas or helpers would be greatly appreciated as especially dealing with webpack or bundling is a dark place for me.

<img width="960" alt="Screenshot 2024-04-12 at 11 26 23" src="https://github.com/enuchi/React-Google-Apps-Script/assets/11017096/fa4ada1e-0e55-4ba0-9cfb-1597bc49e0a2">
<img width="625" alt="Screenshot 2024-04-12 at 11 26 36" src="https://github.com/enuchi/React-Google-Apps-Script/assets/11017096/32110c77-0243-4361-84a3-b0e8cde95612">

**Comment by iamsanteri** [comment]
What makes this interesting is that of the other sample boilerplates app, every one of them works including the MUI dialog-based one except for Bootstrap boilerplate example where it now complains that for whatever reason ReactBootstrap is not found in a similar fashion to the emotionReact in my application: 

<img width="609" alt="Screenshot 2024-04-12 at 11 42 43" src="https://github.com/enuchi/React-Google-Apps-Script/assets/11017096/d1994bb7-7299-419c-a637-441dc75259b1">

What could be the common strand with these that changed and broke the deploy workflow?

**Comment by iamsanteri** [comment]
I've tried continue to look into the issue and when I open the deployed bundle in apps script and try to load emotionReact from unpkg I get a cloudflare error and seems their CDN for this is down, is it possible that this is causing the entire issue?

**Comment by iamsanteri** [comment]
Seems like it's down: https://www.unpkg.com/status/

**Comment by enuchi** [comment]
Interesting -- haven't seen that before. You could try using a different CDN, or you could decide not to externalize all these packages in production, but it may slow down loading the app.

**Comment by iamsanteri** [comment]
I confirm everything works now in production after the CDN is up again. Closing this issue.

**Comment by brandonfake69** [comment]
I'm coming across the same issue in the lastest version and don't think the cdn is a good long-term solution. Like @iamsanteri the base examples work in dev mode and production mode. I then deleted all but the TailwindCSS example and modified that. My modifications worked in dev mode but break in deployment. 
I've also tried upgrading @emotion/react and @emotion/styled to the latest versions (^11.14.0). I've also tried adding unpkg as a cdn:

```
    <script
      crossorigin
      type="text/typescript"
      src="https://unpkg.com/@emotion/react@latest/dist/emotion-react.umd.min.js"
    ></script>
    <script
      crossorigin
      type="text/typescript"
      src="https://unpkg.com/@emotion/styled@latest/dist/emotion-styled.umd.min.js"
    ></script>
```

But get the same error in deployment. This is super weird and I am likewise not sure of why it broke. Almost don't want to reopen the issue since the repo technically works but since both @iamsanteri and I are getting the same issue we can at least have a place to document it when we figure it out (and maybe others who get the error will find this issue).
